### PR TITLE
Error logging: Add future support to rename untrustedOrigin to errorRunningExternalCode

### PR DESCRIPTION
**Pull request by markfields**
## Description

tl;dr - Set us up to rename `untrustedOrigin` logging property to `errorRunningExternalCode` once 2.0 becomes LTS for cross-layer runtime compat.  And make the "read" checks on the truthiness rather than a specific value - I think we may be able to use string values to distinguish errors thrown by a `"dependency"` v. errors thrown by a consumer-provided `"callback"`.

### More context

We put a flag "untrustedOrigin" on error objects instrumented not in FF codebase, to distinguish from our own errors and also highlight the fact that we don't control what info is put in the message.

But it looks mighty strange in the logs when these are logged, especially in the driver layer (e.g. "TypeError: Failed to fetch"). People sometimes worry that this indicates something serious.

Alex and I came up with a more explicit name which is hopefully more clear: "errorRunningExternalCode".  Rather than writing both for the next year+, adding to payload sizes and not actually helping the confusion, I'm suggesting we just plant "read" support for the new name, so we can cut over once 2.0 becomes LTS.

_Note that this flag doesn't mean there's no FF bug, e.g. it could be a TypeError from reading undefined or a JSON parsing error if we screwed up some data format.  But most cases it's from a callback provided by the app._

### CODE REVIEW

**Comment by markfields** [comment]
Chatted offline with @alexvy86 - Will set aside bigger `errorOrigin` or `errorSource` direction if we can come up with a good stand-alone name for this.  Current winner is `errorCallingExternalCode: true` (or `1`)

**Review by alexvy86 — APPROVED** [review]
